callsignlookuptools/callook/callook.py

```diff
@@ -0,0 +1,162 @@
+"""
+callooktools
+---
+Copyright 2021 classabbyamp, 0x5c
+Released under the terms of the BSD 3-Clause license.
+"""
+
+
+from abc import ABC, abstractmethod
+from typing import Optional
+from datetime import datetime
+
+from gridtools import Grid, LatLong
+from pydantic import BaseModel, Field, validator
+
+from ..common import abcs, enums, dataclasses, exceptions
+
+
+class CallookCallsignModel(BaseModel):
+    callsign: Optional[str] = None
+    operClass: enums.LicenseClass = enums.LicenseClass.NONE
+    class_: enums.LicenseClass = Field(default=enums.LicenseClass.NONE, alias="class")
+
+    class Config:
+        anystr_strip_whitespace = True
+        arbitrary_types_allowed = True
+
+
+class CallookTrusteeModel(BaseModel):
+    callsign: Optional[str] = None
+    name: Optional[str] = None
+
+    class Config:
+        anystr_strip_whitespace = True
+        arbitrary_types_allowed = True
+
+
+class CallookAddressModel(BaseModel):
+    line1: Optional[str] = None
+    line2: Optional[str] = None
+    attn: Optional[str] = None
+
+    class Config:
+        anystr_strip_whitespace = True
+        arbitrary_types_allowed = True
+
+
+class CallookLocationModel(BaseModel):
+    latitude: Optional[float] = None
+    longitude: Optional[float] = None
+    gridsquare: Optional[Grid] = None
+
+    @validator("gridsquare", pre=True)
+    def _to_grid(cls, v):
+        if isinstance(v, str):
+            return Grid(v)
+        return v
+
+    class Config:
+        anystr_strip_whitespace = True
+        arbitrary_types_allowed = True
+
+
+class CallookOtherInfoModel(BaseModel):
+    grantDate: Optional[datetime] = None
+    expiryDate: Optional[datetime] = None
+    lastActionDate: Optional[datetime] = None
+    frn: Optional[str] = None
+    ulsUrl: Optional[str] = None
+
+    @validator("grantDate", "expiryDate", "lastActionDate", pre=True)
+    def _parse_date(cls, v):
+        if isinstance(v, str):
+            return datetime.strptime(v, "%m/%d/%Y")
+        return v
+
+    class Config:
+        anystr_strip_whitespace = True
+        arbitrary_types_allowed = True
+
+
+class CallookDataModel(BaseModel):
+    status: enums.CallookStatus = enums.CallookStatus.VALID
+    type: enums.CallsignType = enums.CallsignType.NONE
+    current: CallookCallsignModel = CallookCallsignModel()
+    previous: CallookCallsignModel = CallookCallsignModel()
+    trustee: CallookTrusteeModel = CallookTrusteeModel()
+    name: Optional[str] = None
+    address: CallookAddressModel = CallookAddressModel()
+    location: CallookLocationModel = CallookLocationModel()
+    otherInfo: CallookOtherInfoModel = CallookOtherInfoModel()
```

Comment:
Needs proper default definition for mutable types
https://pydantic-docs.helpmanual.io/usage/models/#field-with-dynamic-default-value